stripe/resource_stripe_promotion_code.go

```diff
@@ -194,20 +205,42 @@ func resourceStripePromotionCodeRead(_ context.Context, d *schema.ResourceData,
 			return nil
 		}(),
 		d.Set("max_redemptions", promotionCode.MaxRedemptions),
-		d.Set("expires_at", time.Unix(promotionCode.ExpiresAt, 0).Format(time.RFC3339)),
+
+		// stripe go client library has ExpiresAt as 0 iff it's not set.
+		func() error {
+			if promotionCode.ExpiresAt != 0 {
+				expiryTime := time.Unix(promotionCode.ExpiresAt, 0)
+				return d.Set("expires_at", expiryTime.Format(time.RFC3339))
+			}
+			return nil
+		}(),
+
 		func() error {
 			if promotionCode.Restrictions != nil {
-				return d.Set("restrictions", []map[string]interface{}{
-					{
-						"first_time_transaction":  promotionCode.Restrictions.FirstTimeTransaction,
-						"minimum_amount":          promotionCode.Restrictions.MinimumAmount,
-						"minimum_amount_currency": promotionCode.Restrictions.MinimumAmountCurrency,
-					},
-				})
+				restrictions := map[string]interface{}{
+					"first_time_transaction": promotionCode.Restrictions.FirstTimeTransaction,
+				}
+
+				// stripe go client library has MinimumAmount as 0 iff it's not set.
+				// stripe API accepts minimum_amount iff > 1 and raise an error otherwise.
+				if minAmount := promotionCode.Restrictions.MinimumAmount; minAmount != 0 {
+					restrictions["minimum_amount"] = minAmount
+					restrictions["minimum_amount_currency"] = promotionCode.Restrictions.MinimumAmountCurrency
+				}
+		
+				if len(restrictions) > 0 {
+					return d.Set("restrictions", []map[string]interface{}{restrictions})
+				}
+			}
+			return nil
+		}(),
+		// only set metadata if it's not empty map
+		func() error {
```

Comment:
Could you please remove this modification?
1. It's not relevant to the issue.
2. It's not consistent with the other resources.
Looks good as an improvement and I would rather see another PR where all resources follow the same pattern.

Reply:
Done!